usage/camp/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/BrooklynDslInterpreter.java

```diff
@@ -168,6 +168,17 @@ public Object evaluateOn(Object o, FunctionWithArgs f, boolean deepEvaluation) {
         
         String fn = f.getFunction();
         fn = Strings.removeFromStart(fn, "$brooklyn:");
+        if (fn.startsWith("function.")) {
+            for (Class<?> c : clazz.getClasses()) {
+                if (c.equals(BrooklynDslCommon.Functions.class)) {
+                    if (o instanceof Class) {
```

Comment:
It's not obvious what you're doing here or why. What do you expect to be passed in (i.e. what is `o`)? Why will `clazz` be a sub-type of (or the concrete type) `BrooklynDslCommon.Functions`?